patches/tModLoader/Terraria/GameContent/ShopHelper.cs.patch

```diff
@@ -140,9 +142,19 @@
 -	public void LoveNPCByTypeName(int npcType)
 +	internal void LoveNPCByTypeName(int npcType)
  	{
- 		AddHappinessReportText("LoveNPC_" + NPCID.Search.GetName(npcType), new {
- 			NPCName = NPC.GetFullnameByID(npcType)
- 		});
++		if (npcType < NPCID.Count)
```

Comment:
needs a type check on this NPC. What if a mod is using this to love another modded NPC? May not be princess.